src/casp/data_manager/base_data_manager.py

```diff
@@ -24,4 +24,4 @@ def __init__(self):
         self.credentials = credentials
         self.project = project
         self.block_coo_matrix_db_client = bigquery.Client(credentials=self.credentials, project=self.project)
-        self.system_data_db_session = init_db()
+        self.system_data_db_session_maker = get_db_session_maker()
```

Comment:
I was thinking about this but I'm not sure.  Each data manager is only initialized once, right?  And then it's used by every thread?  Does that mean there would be only a couple sessions and they'd all be open during the entire time the app is running, or does sqlalchemy do anything special that makes it so that's not the case?

Reply:
I think you're right, yes. I believe we will have to make our service classes a part of fast API dependency injection, so they (and their data managers) get all initialized only once per request. For now your solution seems to be reliable for me.:)